src/index.js

```diff
@@ -74,176 +73,219 @@ addParseCloud();
+    if (process.env.TESTING == 1) {
+      api.use('/', require('./testing-routes').router);
+    }
 
-  api.use(bodyParser.json({ 'type': '*/*' , limit: maxUploadSize }));
-  api.use(middlewares.allowCrossDomain);
-  api.use(middlewares.allowMethodOverride);
-  api.use(middlewares.handleParseHeaders);
+    api.use(bodyParser.json({ 'type': '*/*' , limit: maxUploadSize }));
+    api.use(middlewares.allowCrossDomain);
+    api.use(middlewares.allowMethodOverride);
+    api.use(middlewares.handleParseHeaders);
 
-  let routers = [
-    new ClassesRouter(),
-    new UsersRouter(),
-    new SessionsRouter(),
-    new RolesRouter(),
-    new AnalyticsRouter(),
-    new InstallationsRouter(),
-    new FunctionsRouter(),
-    new SchemasRouter(),
-    new PushRouter(),
-    new LogsRouter(),
-    new IAPValidationRouter(),
-    new FeaturesRouter(),
-  ];
+    let routers = [
+      new ClassesRouter(),
+      new UsersRouter(),
+      new SessionsRouter(),
+      new RolesRouter(),
+      new AnalyticsRouter(),
+      new InstallationsRouter(),
+      new FunctionsRouter(),
+      new SchemasRouter(),
+      new PushRouter(),
+      new LogsRouter(),
+      new IAPValidationRouter(),
+      new FeaturesRouter(),
+    ];
 
-  if (process.env.PARSE_EXPERIMENTAL_CONFIG_ENABLED || process.env.TESTING) {
-    routers.push(new GlobalConfigRouter());
-  }
+    if (process.env.PARSE_EXPERIMENTAL_CONFIG_ENABLED || process.env.TESTING) {
+      routers.push(new GlobalConfigRouter());
+    }
 
-  if (process.env.PARSE_EXPERIMENTAL_HOOKS_ENABLED || process.env.TESTING) {
-    routers.push(new HooksRouter());
-  }
+    if (process.env.PARSE_EXPERIMENTAL_HOOKS_ENABLED || process.env.TESTING) {
+      routers.push(new HooksRouter());
+    }
 
-  let routes = routers.reduce((memo, router) => {
-    return memo.concat(router.routes);
-  }, []);
+    let routes = routers.reduce((memo, router) => {
+      return memo.concat(router.routes);
+    }, []);
 
-  let appRouter = new PromiseRouter(routes);
+    let appRouter = new PromiseRouter(routes);
 
-  batch.mountOnto(appRouter);
+    batch.mountOnto(appRouter);
 
-  api.use(appRouter.expressApp());
+    api.use(appRouter.expressApp());
 
-  api.use(middlewares.handleParseErrors);
+    api.use(middlewares.handleParseErrors);
 
-  process.on('uncaughtException', (err) => {
-    if( err.code === "EADDRINUSE" ) { // user-friendly message for this common error
-      console.log(`Unable to listen on port ${err.port}. The port is already in use.`);
-      process.exit(0);
-    }
-    else {
-      throw err;
+    process.on('uncaughtException', (err) => {
+      if( err.code === "EADDRINUSE" ) { // user-friendly message for this common error
+        console.log(`Unable to listen on port ${err.port}. The port is already in use.`);
+        process.exit(0);
+      }
+      else {
+        throw err;
+      }
+    });
+    this.hooksController.load();
+
+    return api;
```

Comment:
I definitely cannot support turning `ParseServer` into a class. Currently, you can create a Parse Server without using `new` so it would be a compatibility break of something that is pretty major. If `app.use(mountPath, ParseServer({...}))` doesn't work then I will reject the change.